mkosi/installer/dnf.py

```diff
@@ -188,7 +188,7 @@ def cmd(
 
         if not context.config.architecture.is_native():
             cmdline += [
-                f"--forcearch={context.config.distribution.architecture(context.config.architecture)}"
+                f"--forcearch={context.config.distribution.installer.architecture(context.config.architecture)}"
             ]
```

Comment:
This would also be shorter by moving this out
```python
arch = context.config.distribution.installer.architecture(context.config.architecture)
cmdline += [f"--forcearch={arch}]